apm-sniffer/apm-sdk-plugin/xmemcached-2.x-plugin/src/test/java/org/skywalking/apm/plugin/xmemcached/v2/XMemcachedMethodInterceptorTest.java

```diff
@@ -0,0 +1,141 @@
+        when(enhancedInstance.getSkyWalkingDynamicField()).thenReturn("127.0.0.1:11211");
+    } 
+
+    @Test
+    public void testIntercept() throws Throwable {
+        interceptor.beforeMethod(enhancedInstance, getMockSetMethod(), allArgument, argumentType, null);
+        interceptor.afterMethod(enhancedInstance, getMockGetMethod(), allArgument, argumentType, null);
+
+        TraceSegment traceSegment = segmentStorage.getTraceSegments().get(0);
+        List<AbstractTracingSpan> spans = SegmentHelper.getSpans(traceSegment);
+        assertThat(spans.size(), is(1));
+        assertMemcacheSpan(spans.get(0));
+    }
+
+    @Test
+    public void testInterceptWithException() throws Throwable {
+        interceptor.beforeMethod(enhancedInstance, getMockSetMethod(), allArgument, argumentType, null);
+        interceptor.handleMethodException(enhancedInstance, getMockSetMethod(), allArgument, argumentType, new RuntimeException());
+        interceptor.afterMethod(enhancedInstance, getMockSetMethod(), allArgument, argumentType, null);
+
+        TraceSegment traceSegment = segmentStorage.getTraceSegments().get(0);
+        List<AbstractTracingSpan> spans = SegmentHelper.getSpans(traceSegment);
+        assertThat(spans.size(), is(1));
+        assertMemcacheSpan(spans.get(0));
+
+        assertLogData(SpanHelper.getLogs(spans.get(0)));
+    }
+
+    private void assertLogData(List<LogDataEntity> logDataEntities) {
+        assertThat(logDataEntities.size(), is(1));
+        LogDataEntity logData = logDataEntities.get(0);
+        Assert.assertThat(logData.getLogs().size(), is(4));
+        Assert.assertThat(logData.getLogs().get(0).getValue(), CoreMatchers.<Object>is("error"));
+        Assert.assertThat(logData.getLogs().get(1).getValue(), CoreMatchers.<Object>is(RuntimeException.class.getName()));
+        Assert.assertNull(logData.getLogs().get(2).getValue());
+        assertNotNull(logData.getLogs().get(3).getValue());
+    }
+
+    private void assertMemcacheSpan(AbstractTracingSpan span) {
+        assertThat(span.getOperationName(), is("XMemcached/set"));
+        assertThat(span.isExit(), is(true));
+        assertThat(SpanHelper.getComponentId(span), is(20));
+        List<KeyValuePair> tags = SpanHelper.getTags(span);
+        assertThat(tags.get(0).getValue(), is("Memcache"));
+        assertThat(tags.get(1).getValue(), is("set OperationKey"));
+        assertThat(SpanHelper.getLayer(span), is(SpanLayer.DB));
+    }
+    
+    private Method getMockSetMethod() {
+        try {
+            return XMemcachedClient.class.getMethod("set", String.class, int.class, Object.class);
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+    
+    private Method getMockGetMethod() {
+        try {
+            return XMemcachedClient.class.getMethod("get", String.class);
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
```

Comment:
remove e.printStackTrace()

Reply:
In a test case, I think you can remove the try/catch instead of `e.printStackTrace`. The error should not happen if the test case is right.